### ISSUE TITLE
Wrong folder/filename givin in title bar when showing files from the .leo folder

### ISSUE THREAD

**Issue by boltex** [leo-editor/leo-editor#3738]
To reproduce, open the workbook.leo in the .leo folder of your user on your windows machine (usually in C:\Users\YOURNAME\.leo )

I think it grabs the first ".leo" it finds instead of the last and therefore thinks the filename is the last part of the path as shown [here](https://github.com/leo-editor/leo-editor/assets/847716/e348ba0a-9e08-4cab-88cb-cbed992913d0).

**Comment by tbpassin** [comment]
I just tried this and got the same ".leo" in the title bar after I opened another Leo outline in the .leo folder.  But then I opened another instance of Leo and repeated the test and did not get the glitch.

So the problem may be a little more subtle, but I definitely see it sometimes.

**Comment by boltex** [comment]
@tbpassin Thanks for this reproduction experiment! 

Sometimes, subtle details like the one you've just spotted help a lot in pin-pointing the actual bug & fixing it.

**Comment by boltex** [comment]
@tbpassin @edreamleo I've done more tests to illustrate the problem more precisely and reproduce the bug reliably:

Here, After doing 'save as' on testMd1 renaming it testMd2, and then re-opening testMd1 from the recent files menu, I get those two files opened from the same folder, but with title bar somewhat tangled/wrong for testMd2, ( for comparison, the second tab for testMd1 below is ok)
![image](https://github.com/leo-editor/leo-editor/assets/847716/e9373748-7a39-437a-be29-bf776087e5aa)

![image](https://github.com/leo-editor/leo-editor/assets/847716/b9e72b72-5395-4f8a-ab99-dab3aa87949a)

**Comment by edreamleo** [comment]
@boltex @tbpassin This issue is a tiny nit.

Are either of you interested in handling it? I'm not.

**Comment by edreamleo** [comment]
@tbpassin Thanks for volunteering!

**Comment by tbpassin** [comment]
Very strange.  I reproduced my case where the filename in the title was only ".leo", then ran c.computeWindowTitle() in that outline and it returned the correct name.

**Comment by tbpassin** [comment]
So far the only case where I can reproduce this bug is in a particular Leo window I've had open for days.  I have several outlines open in it.  If I open another outline, then I see the bug.  But when I open another Leo window and open the same outlines, then I don't get the bug.  I'm going to close that Leo window and see if re-opening it will still exhibit the same bug.

**Comment by tbpassin** [comment]
After closing and re-opening that Leo instance, I didn't get the bug.  That's going to make it awfully hard to debug...

**Comment by edreamleo** [comment]
@tbpassin Thanks for your testing. There is no rush and you can always say no to this issue.

**Comment by tbpassin** [comment]
> I've done more tests to illustrate the problem more precisely and reproduce the bug reliably:


> - Do 'save as' on testMd1.
> - Rename it to testMd2.
> - Reopen testMd1 from the recent files menu

> Now I get those two files opened from the same folder, but with title bar somewhat tangled/wrong for testMd2 as shown [here](https://github.com/leo-editor/leo-editor/assets/847716/e9373748-7a39-437a-be29-bf776087e5aa).

Following this I was able to cause the same failure ... until the next time I restarted the session.  After that, the titles displayed correctly.

**Comment by tbpassin** [comment]
When this happens, the title actually displayed does not match the return from `c.computeWindowTitle()`.

**Comment by edreamleo** [comment]
@tbpassin Good work!

**Comment by tbpassin** [comment]
If I do a save on the outline, the window title changes back to the correct one. If I tab away to another outline and then tab back to the first, the erroneous title shows up again.

It is as if some other method computes the spurious window title, because I haven't been able to coerce `c.computeWindowTitle()` to misbehave like that. Tabbing back to the outline uses the other method, saving uses `c.computeWindowTitle()` to do the computation.

**Comment by tbpassin** [comment]
OK, I know what the bug is although I don't know where it's happening.  The spurious title is created when  you feed the correct title to `c.computeWindowTitle()`.  This apparently happens when the outline gets re-opened in the test case but not when the outline as saved.

As a reminder, here is the test case:
1. Save outline with new name.  To be definite, call the original outline _c:\temp\leo\test1.leo_.  Save it as _c:\temp\leo\test2.leo_.
2. Observer there is one open tab.  The Leo window has the title _test2.leo in c:\test\leo_.
3. Re-open test1.leo, either from the _Recent Files_ menu or the _Open-File_ menu.
4. Switch to the newly-reopened outline by clicking its tab.
5. Observer that the window title is now the incorrect _leo in test-2.leo in c:\temp_

**Comment by tbpassin** [comment]
I found where this is happening. It's in `frameFactory.signal handlers` `slotCurrentChanged()`.  Now that I know what to look for, it's right there:
```
    title = c.computeWindowTitle(f.title)
    tabw.setWindowTitle(title)
```
We are giving the title to `c.computeWindowTitle()` but we need to be giving the outline file's path instead.

**Comment by tbpassin** [comment]
And.... the bug is gone!  I should just check some more cases to make sure that nothing else has been messed up before a PR.

**Comment by tbpassin** [comment]
I see one little anomaly.  Going through the procedure that displayed the bug, the letter "c" in _C:\temp..._  is lower case, and that's how it is in the Recent Files list, too.  All the others have capital "C".

I'm not inclined to fuss with this quirp, and it must happen somewhere else besides this fix, I would think.

**Comment by edreamleo** [comment]
@tbpassin Excellent work. I've noticed the problem with the case of the drive letter in other contexts.

I agree that the case of the drive letter is a separate issue/quirp.

**Comment by edreamleo** [comment]
Fixed via PR #3742.
